core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAO.scala

```diff
@@ -173,4 +194,30 @@ class BillingProfileManagerDAOImpl(
       )
 
   override def getStatus(): SystemStatus = apiClientProvider.getUnauthenticatedApi().serviceStatus()
+
+  @throws(classOf[BpmAzureSpendReportBadRequest])
+  def getAzureSpendReport(billingProfileId: UUID,
+                          spendReportStartDate: Date,
+                          spendReportEndDate: Date,
+                          ctx: RawlsRequestContext
+  ): SpendReport =
+    try
+      apiClientProvider
+        .getSpendReportingApi(ctx)
+        .getSpendReport(
+          billingProfileId,
+          spendReportStartDate,
+          spendReportEndDate
+        )
+    catch {
+      case ex: ApiException =>
+        if (ex.getCode == StatusCodes.BadRequest.intValue) {
```

Comment:
It could be good to make sure all BPM exceptions are getting propagated up to the user correctly, not just 400s.

Reply:
Why is BadRequest handled in a special way? A comment about the use case could be helpful.

Reply:
To parse response from BPM and return validation message

Reply:
And that is the only case where the response is JSON?

Reply:
This is the only case where I need to extract message, otherwise I return ErrorReport with the whole exception